opts/mod.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2025 Synadia Communications, Inc
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ConnectionOptions } from "@nats-io/nats-core";
+import { wsconnect } from "@nats-io/nats-core";
+
+type ConnectionProperty = Omit<
+  keyof ConnectionOptions & { creds: string },
+  "authenticator" | "reconnectDelayHandler"
+>;
+
+const booleanProps = [
+  "debug",
+  "ignoreAuthErrorAbort",
+  "ignoreClusterUpdates",
+  "noAsyncTraces",
+  "noEcho",
+  "noRandomize",
+  "pedantic",
+  "reconnect",
+  "resolve",
+  "tls",
+  "verbose",
+  "waitOnFirstConnect",
+] as const;
+
+const numberProps = [
+  "maxPingOut",
+  "maxReconnectAttempts",
+  "pingInterval",
+  "reconnectJitter",
+  "reconnectJitterTLS",
+  "reconnectTimeWait",
+  "port",
+  "timeout",
+] as const;
+
+const stringProps = [
+  "name",
+  "pass",
```

Comment:
Should name and pass really be allowed as query parameters, instead of being forced to be in the authority part of the URL?  Allowing them to leak out of authority means that anything redacting URLs before logging them needs to be aware of this scheme and which parts might be secret.  Whereas anything which understands the authority part of most common URL schemes will redact without special support.

Reply:
It is not an URL, it is a string, that uses queryparams type of thing.
```
nats-opts:servers=wss%253A%252F%252Fdemo.nats.io%253A8443&name=me&timeout=10000&debug=true&noEcho=true&reconnect=false&handshakeFirst=true
```

Have we tried that with nats:// urls? On environments like JavaScript that protocol is invalid and won't parse.
With that said, we can force it into something like that, but then the problem with having it be an URL is that the protocol will define a default port - so parsing it could introduce ports that are not expected, but would have to verify. We can encode as URL and if doesn't work, we can make it be an exercise to the user

Reply:
@philpennock have a look now - the main issue with this is that now the URL looks like `nats://` or `natss://` which will create ambiguity on whether this is something that can be passed directly to the client.

Reply:
So I _still_ don't see an updated PR for this "have a look now" ... by any chance, did you update and commit locally and not push to GitHub?